frontend/app/routes/public/renew/$id/ita/update-home-address.tsx

```diff
@@ -73,63 +73,63 @@ export async function action({ context: { appContainer, session }, params, reque
+        } else if (!isValidPostalCode(val.country, val.postalCode)) {
+          const message = val.country === CANADA_COUNTRY_ID ? t('renew-ita:update-address.error-message.home-address.postal-code-valid') : t('renew-ita:update-address.error-message.home-address.zip-code-valid');
           ctx.addIssue({ code: z.ZodIssueCode.custom, message, path: ['homePostalCode'] });
-        } else if (val.homeCountry === CANADA_COUNTRY_ID && val.homeProvince && !isValidCanadianPostalCode(val.homeProvince, val.homePostalCode)) {
+        } else if (val.country === CANADA_COUNTRY_ID && val.province && !isValidCanadianPostalCode(val.province, val.postalCode)) {
           ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('renew-ita:update-address.error-message.home-address.invalid-postal-code-for-province'), path: ['homePostalCode'] });
         }
       }
 
-      if (val.homeCountry && val.homeCountry !== CANADA_COUNTRY_ID && val.homePostalCode && isValidPostalCode(CANADA_COUNTRY_ID, val.homePostalCode)) {
+      if (val.country && val.country !== CANADA_COUNTRY_ID && val.postalCode && isValidPostalCode(CANADA_COUNTRY_ID, val.postalCode)) {
         ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('renew-ita:update-address.error-message.home-address.invalid-postal-code-for-country'), path: ['homeCountry'] });
       }
     })
     .transform((val) => ({
       ...val,
-      homePostalCode: val.homeCountry && val.homePostalCode ? formatPostalCode(val.homeCountry, val.homePostalCode) : val.homePostalCode,
-    })) satisfies z.ZodType<AddressInformationState>;
+      postalCode: val.country && val.postalCode ? formatPostalCode(val.country, val.postalCode) : val.postalCode,
+    })) satisfies z.ZodType<HomeAddressState>;
 
-  const parsedDataResult = addressInformationSchema.safeParse({
-    homeAddress: formData.get('homeAddress') ? String(formData.get('homeAddress')) : undefined,
-    homeCountry: formData.get('homeCountry') ? String(formData.get('homeCountry')) : undefined,
-    homeProvince: formData.get('homeProvince') ? String(formData.get('homeProvince')) : undefined,
-    homeCity: formData.get('homeCity') ? String(formData.get('homeCity')) : undefined,
-    homePostalCode: formData.get('homePostalCode') ? String(formData.get('homePostalCode')) : undefined,
+  const parsedDataResult = homeAddressSchema.safeParse({
+    address: formData.get('homeAddress') ? String(formData.get('homeAddress')) : undefined,
+    country: formData.get('homeCountry') ? String(formData.get('homeCountry')) : undefined,
+    province: formData.get('homeProvince') ? String(formData.get('homeProvince')) : undefined,
+    city: formData.get('homeCity') ? String(formData.get('homeCity')) : undefined,
+    postalCode: formData.get('homePostalCode') ? String(formData.get('homePostalCode')) : undefined,
   });
```

Comment:
Convention is to not set mandatory fields as `undefined`:
```suggestion
  const parsedDataResult = homeAddressSchema.safeParse({
    address: String(formData.get('homeAddress')),
    country: String(formData.get('homeCountry')),
    province: formData.get('homeProvince') ? String(formData.get('homeProvince')) : undefined,
    city: String(formData.get('homeCity')) ,
    postalCode: formData.get('homePostalCode') ? String(formData.get('homePostalCode')) : undefined,
  });
```